src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,153 @@
+package duke.task;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+import duke.command.MalformedCommandException;
+import duke.storage.StorageException;
+
+/**
+ * Represents a collection of all the user's tasks.
+ */
+public class TaskList {
+    private List<Task> tasks;
+
+    /**
+     * Creates the task list.
+     */
+    public TaskList() {
+        tasks = new ArrayList<>();
+    }
+
+    /**
+     * Loads task from file on disk into the task list.
+     *
+     * @param file File storing tasks on disk.
+     * @throws StorageException If file storing the tasks is incorrectly formatted.
+     */
+    public void loadTasks(File file) throws StorageException {
```

Comment:
Perhaps storage should be in charge of reading the data from the file? I think it would be better if Storage passes an ArrayList of tasks for TaskList to initialise